install_script_op_worker.sh

```diff
@@ -0,0 +1,553 @@
+    no_start=true
+  fi
+fi
+
+if [ "$site" ]; then
+  printf "\033[34m\n* Setting SITE in the $nice_flavor configuration: $config_file\n\033[0m\n"
+  $sudo_cmd sh -c "sed -i 's/# site:.*/site: $site/' $config_file"
+fi
+
+if [ ! -e "$etcdir"/pipeline.yaml ]; then
+  printf "\033[31mThe $nice_flavor won't start automatically at the end of the script because the pipeline configuration is missing,\nplease add one at $etcdir/pipeline.yaml and start the $nice_flavor manually.\n\033[0m\n"
+  no_start=true
+fi
+
+$sudo_cmd chown observability-pipelines-worker:observability-pipelines-worker "$config_file"
+$sudo_cmd chmod 640 "$config_file"
+
+# Creating or overriding the install information
+install_info_content="---
+install_method:
+  tool: install_script
+  tool_version: install_script_op_worker
+  installer_version: install_script-$install_script_version
+"
+
+$sudo_cmd sh -c "echo '$install_info_content' > $etcdir/install_info"
+
+service_cmd="service"
+
+if [ $no_start ]; then
+  printf "\033[34m\n  * DD_INSTALL_ONLY environment variable set.\033[0m\n"
+fi
+
+# Use /usr/sbin/service by default.
+# Some distros usually include compatibility scripts with Upstart or Systemd. Check with: `command -v service | xargs grep -E "(upstart|systemd)"`
+restart_cmd="$sudo_cmd $service_cmd $worker_flavor restart"
+stop_instructions="$sudo_cmd $service_cmd $worker_flavor stop"
+start_instructions="$sudo_cmd $service_cmd $worker_flavor start"
+
+if [[ `$sudo_cmd ps --no-headers -o comm 1 2>&1` == "systemd" ]] && command -v systemctl 2>&1; then
+  # Use systemd if systemctl binary exists and systemd is the init process
+  restart_cmd="$sudo_cmd systemctl restart ${worker_flavor}.service"
+  stop_instructions="$sudo_cmd systemctl stop $worker_flavor"
+  start_instructions="$sudo_cmd systemctl start $worker_flavor"
+elif /sbin/init --version 2>&1 | grep -q upstart; then
+  # Try to detect Upstart, this works most of the times but still a best effort
+  restart_cmd="$sudo_cmd stop $worker_flavor || true ; sleep 2s ; $sudo_cmd start $worker_flavor"
+  stop_instructions="$sudo_cmd stop $worker_flavor"
+  start_instructions="$sudo_cmd start $worker_flavor"
+fi
+
+if [ $no_start ]; then
+  printf "\033[34m\n    The newly installed version of the ${nice_flavor} will not be started.
+  You will have to do it manually using the following command:
+
+  $start_instructions\033[0m\n\n"
+
+else
+  printf "\033[34m* Starting the ${nice_flavor}...\n\033[0m\n"
+  ERROR_MESSAGE="Error starting ${nice_flavor}"
+
+  eval "$restart_cmd"
+
+  ERROR_MESSAGE=""
+    
+  # Metrics are submitted, echo some instructions and exit
+  printf "\033[32m  Your ${nice_flavor} is running and functioning properly.\n\033[0m"
+
+  printf "\033[32m  It will continue to run in the background and submit metrics to Datadog.\n\033[0m"
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏ This doesn't seem right :)

Reply:
_Technically_ it's true, OPW reports it's internal metrics and logs to Datadog - but I can drop this line if it's considered more wrong than right.